docker-compose.production.yml

```diff
@@ -30,4 +38,6 @@ services:
   redis:
     image: redis:6.2-alpine
   webapp:
+    depends_on:
+      - api
```

Comment:
Why did the webapp not have a dependency on the api service before?

Reply:
To be fair, since this compose file is being deployed as a Stack in a Swarm, the `depends_on` config will be ignored. Maybe all the `depends_on` should just be removed since they don't do anything.

While I was updating the file, I put it in there out of habit, since this compose file can be use with Docker Compose as well as Docker Swarm. That's speculative generalization, though.

Reply:
Would it be valuable/necessary to find a solution that'll allow any dependencies to be expressed?

Reply:
The lack of reliability caused by not defining version and service dependencies is within our tolerance. There are some situations that we can avoid by having app code check versions and availability of deployed dependencies, but the app isn't used enough to make that work worth the effort.

Reply:
Thanks for explaining.  

When you say "not defining version" are you referring to Docker compose version property defined in a Docker Compose file and that it is only "informative" and is deprecated? Are are you referring to versioning of the app itself?

What usage metrics would be useful to monitor in order to measure when we would need to have app code check versions and availability of deployed dependencies?

Reply:
Thanks for the explanation!